hnn_core/gui.py

```diff
@@ -0,0 +1,636 @@
+                variables['net'].cell_response.plot_spikes_raster(ax=ax)
+            else:
+                print("No data")
+
+        elif plot_type['new'] == 'current dipole':
+            if variables['dpls'] is not None:
+                variables['dpls'][0].plot(ax=ax)
+            else:
+                print("No data")
+
+        elif plot_type['new'] == 'input histogram':
+            # BUG: got error here, need a better way to handle exception
+            if variables['net'] is not None and sum(
+                [len(_)
+                 for _ in variables['net'].cell_response._spike_times]) > 0:
+                variables['net'].cell_response.plot_spikes_hist(ax=ax)
+            else:
+                print("No data")
+
+        elif plot_type['new'] == 'PSD':
+            if variables['dpls'] is not None:
+                variables['dpls'][0].plot_psd(fmin=0, fmax=50, ax=ax)
+            else:
+                print("No data")
+
+        elif plot_type['new'] == 'spectogram':
+            freqs = np.arange(10., 100., 1.)
+            n_cycles = freqs / 8.
+            if variables['dpls'] is not None:
+                variables['dpls'][0].plot_tfr_morlet(freqs,
+                                                     n_cycles=n_cycles,
+                                                     ax=ax)
+            else:
+                print("No data")
+
+        elif plot_type['new'] == 'network':
+            if variables['net'] is not None:
+                variables['net'].plot_cells(ax=ax)
+            else:
+                print("No data")
+
+
+def on_upload_change(change, sliders, params, tstop, tstep, log_out):
+    if len(change['owner'].value) == 0:
+        return
+
+    params_fname = change['owner'].metadata[0]['name']
+    file_uploaded = change['owner'].value
+    param_data = list(file_uploaded.values())[0]['content']
+    param_data = codecs.decode(param_data, encoding="utf-8")
+
+    ext = op.splitext(params_fname)[1]
+    read_func = {'.json': _read_json, '.param': _read_legacy_params}
+    params_network = read_func[ext](param_data)
+
+    log_out.clear_output()
+    with log_out:
+        print(f"parameters: {params_network.keys()}")
+
+    for slider in sliders:
+        for sl in slider:
+            key = 'gbar_' + sl.description
+            sl.value = params_network[key]
+
+    if 'tstop' in params_network.keys():
+        tstop.value = params_network['tstop']
+    if 'dt' in params_network.keys():
+        tstep.value = params_network['dt']
+
+    params.update(params_network)
+
+
+def run_button_clicked(log_out, plot_out, drive_widgets, variables, tstep,
+                       tstop, ntrials, mpi_cmd, params, b):
+    """Run the simulation and plot outputs."""
+    plot_out.clear_output()
+    log_out.clear_output()
+    with log_out:
+        params['dt'] = tstep.value
+        params['tstop'] = tstop.value
+        variables['net'] = Network(params, add_drives_from_params=False)
+
+    try:
```

Comment:
this seems rather dangerous ... I would not put things in `try` blocks as it would make it hard to debug user problems.